time_interp/include/time_interp_external2.inc

```diff
@@ -20,14 +20,38 @@
 !> @addtogroup time_interp_external2_mod
 !> @{
 
+    !> @brief 1D interpolation for @ref time_interp_external
+    subroutine TIME_INTERP_EXTERNAL_1D_(index, time, data_in, verbose,horz_interp, mask_out, &
+               is_in, ie_in, window_id)
+
+      integer, intent(in) :: index
+      type(time_type), intent(in) :: time
+      real(FMS_TI_KIND_), dimension(:), intent(inout) :: data_in
+      logical, intent(in), optional :: verbose
+      type(horiz_interp_type),intent(in), optional :: horz_interp
+      logical, dimension(:), intent(out), optional :: mask_out !< set to true where output data is valid
+      integer, intent(in), optional :: is_in, ie_in
+      integer, intent(in), optional :: window_id
+
+      real(FMS_TI_KIND_), dimension(size(data_in,1), 1, 1) :: data_out
+      logical, dimension(size(data_in), 1, 1) :: mask3d
+
+      data_out(:,1,1) = data_in(:) ! fill initial values for the portions of array that are not touched by 3d routine
+      call time_interp_external(index, time, data_out, verbose, horz_interp, mask3d, &
+                                is_in=is_in,ie_in=ie_in,js_in=1,je_in=1,window_id=window_id)
+      data_in(:) = data_out(:,1,1)
+      if (PRESENT(mask_out)) mask_out(:) = mask3d(:,1,1)
+
+    end subroutine TIME_INTERP_EXTERNAL_1D_
+
+
     !> @brief 2D interpolation for @ref time_interp_external
-    subroutine TIME_INTERP_EXTERNAL_2D_(index, time, data_in, interp, verbose,horz_interp, mask_out, &
```

Comment:
Maybe i am missing something, but why is this being removed? It is just not used?

Reply:
It's not used, it might be easier to follow in this separate cleanup PR https://github.com/NOAA-GFDL/FMS/pull/1762/files

Reply:
How did you determine what's used and what isn't in this context?

Reply:
I followed the variable.  Ultimately, interp is used to set the variable interp_method but interp_method is not used anywhere.

Reply:
What we need to also do is look through the recent component models and FMS codebase to ensure this optional argument is not present for any of the callers.  If it is, we need to preserve the option, but make it clear this option exists only for API compatibility.